en/micro-integrator/docs/setup/deployment/kubernetes_deployment_patterns.md

```diff
@@ -0,0 +1,89 @@
+# Kubernetes Deployment Patterns
+
+!!! Warning
+	**This content is currently work in progress**
+
+These are the deployment patterns you can use when deploying your WSO2 Micro Integrator-based integration solutions in a Kubernetes environment. 
+
+<!--
+The following Kubernetes concepts 
+
+-   Kuberentes cluster
+-   Worker nodes
+-   Pods
+-   Replicas
+-   High availablity in Kuberenetes
+-   Load balancing in Kuberentes
+-->
+
+When you deploy your integrations, the main concern is to ensure high availability and scalability of your system. Therefore, you need to decide upon the number of worker nodes and the number of replicas that are required to scale the deployment and to ensure high availability. 
+
+## Single Replica
+
+The following diagram depicts a single worker node deployment, which contains a single pod (single replica).
+
+<img src="../../../assets/img/k8s_deployment/k8s-single-pod.png">
+
+Designing with a single-node will give you less management overhead if you are building an on-premises cluster (this does not apply to cloud instances). Also, there will be lower cost and resource requirements when compared to a multiple node cluster. 
+
+### Load balancing
+
+This single replica deployment is not recommended if you expect a high amount of traffic to your deployment. With only one replica, it is not possible to balance a high workload across multiple instances.
+
+### High availability
+
+The failure of this single worker node will take down the entire Micro Integrator deployment (depends on how many resources you have additionally). Also, if there is a failure in the single pod, there will be a downtime before a new pod is spawned again. Therefore, to achieve high availability, you require more than one worker node (which avoids single point of failure) and multiple replicas of the pod (which avoids downtime).
+
+### Startup time
+
+The server start-up time is 10s. Therefore, this type of deployment can have a minimum downtime of 10s (either during a rollout update or during a pod failure). This is the biggest disadvantage of this approach since this downtime can vary upon the time it takes to deploy all the capps during the server startup, which could result in a significant downtime in your cluster. If this downtime is acceptable and does not negatively impact your requirements, this approach may be the easiest way for you to get started.
```

Comment:
Shall we not specify the startup time as 10s here since it can sometimes vary upon different facts such as the underline OS and etc. Also, without having a separate title as startup time, I think we can summarize the two paras under HA to a something like below, since it is related to HA. Please change the wordings appropriately.

"The failure of this single worker node will take down the entire Micro Integrator deployment (depends on how many resources you have additionally). Also, if there is a failure in the single pod, there will be a downtime before a new pod is spawned again. This downtime can vary depending on the server_startup_time + artifact_deployment_time, which could lead to a significant downtime in your cluster. If this downtime is acceptable and does not negatively impact your requirements, this approach may be the easiest way for you to get started. Otherwise, to achieve high availability, you require more than one worker node (which avoids single point of failure) and multiple replicas of the pod (which avoids downtime)."